File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/api/follow/FollowController.java
package io.dustin.apps.board.api.follow;

import io.dustin.apps.board.api.usecase.follow.DeleteFollowUseCase;
import io.dustin.apps.board.api.usecase.follow.WriteFollowUseCase;
import io.dustin.apps.board.domain.follow.model.dto.FollowDto;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1")
@RequiredArgsConstructor
public class FollowController {
    private final WriteFollowUseCase writeFollowUseCase;
    private final DeleteFollowUseCase deleteFollowUseCase;

    @PostMapping("follow/{followingId}")
    public FollowDto follow(@PathVariable("followingId") Long followingId, @RequestBody FollowDto followDto) {
        /**
         * {id} 에 해당하는 유저 팔로워 증가 및 로그인 유저 팔로잉 증가 로직 필요
         * */
        return writeFollowUseCase.execute(followingId, followDto.followerID());

    }

    @DeleteMapping("unfollow/{followingId}")
    public FollowDto unfollow(@PathVariable("followingId") Long followingId, @RequestBody FollowDto followDto) {
        /**
         * {id} 에 해당하는 유저 팔로워 감소 및 로그인 유저 팔로잉 감소 로직 필요
         * */
        return deleteFollowUseCase.execute(followingId, followDto.followerID());

    }


}

File: apps/src/main/kotlin/io/dustin/apps/board/api/follow/FollowController.kt
package io.dustin.apps.board.api.follow

import io.dustin.apps.board.api.follow.request.command.FollowCommand
import io.dustin.apps.board.api.usecase.follow.DeleteFollowUseCase
import io.dustin.apps.board.api.usecase.follow.WriteFollowUseCase
import io.dustin.apps.board.domain.follow.model.dto.FollowDto
import io.dustin.apps.common.code.CommonMessage
import io.dustin.apps.common.model.response.CommonResponse
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import java.time.LocalDateTime


@RestController
@RequestMapping("/api/v1/follow")
@Tag(name = "팔로우 관련 API", description = "팔로우 관련 API를 제공한다.")
class FollowController (

    private val writeFollowUseCase: WriteFollowUseCase,
    private val deleteFollowUseCase: DeleteFollowUseCase,

    ) {

    @ResponseStatus(HttpStatus.OK)
    @PostMapping("following")
    @Operation(
        summary = "팔로우 추가",
        description = "팔로우 추가"
    )
    fun follow(@RequestBody @Valid command: FollowCommand): CommonResponse {

        writeFollowUseCase.execute(command.followingId, command.followerId)
        return CommonResponse(
            code = HttpStatus.OK.value(),
            message = CommonMessage.SUCCESS.code,
            timestamp = LocalDateTime.now()
        )
    }

    @ResponseStatus(HttpStatus.OK)
    @DeleteMapping("unfollow")
    @Operation(
        summary = "팔로우 취소",
        description = "팔로우 취소"
    )
    fun unfollow(@RequestBody @Valid command: FollowCommand): CommonResponse {

        deleteFollowUseCase.execute(command.followingId, command.followerId)
        return CommonResponse(
            code = HttpStatus.OK.value(),
            message = CommonMessage.SUCCESS.code,
            timestamp = LocalDateTime.now()
        )
    }
}
